File: app/schemas/journal.py
"""Schemas for trading journal endpoints."""
from __future__ import annotations

from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class JournalEntryResponse(BaseModel):
    id: str
    symbol: str
    interval: str
    signal: str
    lstm_signal: Optional[str]
    patchtst_signal: Optional[str]
    fusion_score: Optional[float]
    entry_price: Optional[float]
    take_profit_1: Optional[float]
    take_profit_2: Optional[float]
    take_profit_3: Optional[float]
    stop_loss: Optional[float]
    confidence: Optional[float]
    status: str
    outcome: Optional[str]
    exit_price: Optional[float]
    pips: Optional[float]
    created_at: datetime
    verified_at: Optional[datetime]

    model_config = {"from_attributes": True}


class JournalStatsResponse(BaseModel):
    total_trades: int
    wins: int
    losses: int
    pending: int
    win_rate: float
    profit_factor: float
    total_pips: float
    max_drawdown: float


class JournalListResponse(BaseModel):
    entries: list[JournalEntryResponse]
    stats: JournalStatsResponse
    page: int
    page_size: int
    total: int

File: fusion-fronted/src/types/journal.ts
export type JournalStatus = 'PENDING' | 'TP1_HIT' | 'TP2_HIT' | 'TP3_HIT' | 'SL_HIT' | 'EXPIRED' | 'IGNORED'
export type JournalOutcome = 'WIN' | 'LOSS' | 'NEUTRAL' | null

export interface JournalEntry {
  id: string
  symbol: string
  interval: string
  signal: string
  lstm_signal: string | null
  patchtst_signal: string | null
  fusion_score: number | null
  entry_price: number | null
  take_profit_1: number | null
  take_profit_2: number | null
  take_profit_3: number | null
  stop_loss: number | null
  confidence: number | null
  status: JournalStatus
  outcome: JournalOutcome
  exit_price: number | null
  pips: number | null
  created_at: string
  verified_at: string | null
}

export interface JournalStats {
  total_trades: number
  wins: number
  losses: number
  pending: number
  win_rate: number        // 0–100
  profit_factor: number
  total_pips: number
  max_drawdown: number
}

export interface JournalResponse {
  entries: JournalEntry[]
  stats: JournalStats
  page: number
  page_size: number
  total: number
}
